server/quota/quota_manager.go

```diff
@@ -0,0 +1,155 @@
+package quota
+
+import (
+	"context"
+	"math"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/quota/config"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/throttled/throttled/v2"
+	"github.com/throttled/throttled/v2/store/goredisstore.v8"
+
+	qcpb "github.com/buildbuddy-io/buildbuddy/proto/quota_config"
+)
+
+const (
+	maxRedisRetry = 3
+)
+
+type namespace struct {
+	name string
+
+	defaultRateLimiter *throttled.GCRARateLimiterCtx
+
+	rateLimiters map[string]*throttled.GCRARateLimiterCtx
+
+	rateLimitersByKey map[string]*throttled.GCRARateLimiterCtx
```

Comment:
Probably worth adding a comment distinguishing the purposes of these two maps, particularly considering that I couldn't find where `rateLimitersByKey` is populated or `rateLimiters` is retrieved from. Are these meant to be the same map?

Reply:
not relevant anymore.